tools/data-handler/src/macros/createCards/index.ts

```diff
@@ -27,6 +27,15 @@ class CreateCardsMacro extends BaseMacro {
   constructor() {
     super(macroMetadata);
   }
+
+  handleValidate = (context: MacroGenerationContext, data: string) => {
+    if (!data || typeof data !== 'string') {
+      throw new Error('createCards macro requires a JSON object as data');
+    }
```

Comment:
handleInject could call this.handleValidate instead of repeating the logic twice

Reply:
I added a private validate() -function that can be reused in different modes. Validation returns a class-specific `options` object so I did not do this at the class interface level.